teuthology/repo_utils.py

```diff
@@ -262,12 +268,13 @@ def fetch_branch(repo_path, branch, shallow=True):
             raise GitError("git fetch failed!")
 
 
-def reset_repo(repo_url, dest_path, branch):
+def reset_repo(repo_url, dest_path, branch, commit=None):
     """
 
     :param repo_url:  The full URL to the repo (not including the branch)
     :param dest_path: The full path to the destination directory
     :param branch:    The branch.
+    :param commit:    The commit to checkout, if different from HEAD.
```

Comment:
- Add the defaults info.
- Is it checked if the commit different from the HEAD indeed, if not, why this message here?